### PR TITLE
Clarify ready

### LINKED ISSUE #38: DeploymentStatus readyReplicas docstring needs clarification.
https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L411

The docstring defines `readyReplicas` as "Total number of ready pods targeted by this deployment" which is defining "ready" in terms of being "ready".  Please clarify the docstring on what "ready" means in this context since the docstring is used for the user-facing docs at https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/deployment-v1/.  

Fixing this will also resolve https://github.com/kubernetes/website/issues/28316

### PR DESCRIPTION
**Pull request by matthiasgubler**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Clarifies and aligns the ready definition for the DeploymentStatus, DeploymentStatus, ReplicationControllerStatus, StatefulSetStatus, and DaemonSetStatus

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/api/issues/38

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

### CODE REVIEW

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 1]
```diff
@@ -237,7 +237,7 @@ type StatefulSetStatus struct {
 	// replicas is the number of Pods created by the StatefulSet controller.
 	Replicas int32 `json:"replicas" protobuf:"varint,2,opt,name=replicas"`
 
-	// readyReplicas is the number of Pods created by the StatefulSet controller that have a Ready Condition.
+	// Total number of pods created by this StatefulSet controller with a Ready Condition by passing the readinessProbe.
```
Comment (liggitt) [inline staging/src/k8s.io/api/apps/v1/types.go:240]: comment should start with the serialized field name (`readyReplicas`)

applies to all modified godoc lines

Comment (liggitt) [reply staging/src/k8s.io/api/apps/v1/types.go:240]: Does this accurately describe behavior for pods without readiness probes? What about pods that set spec.readinessGates?

Referencing the Ready condition makes sense... I'm not sure denormalizing all the inputs that go into determining the Ready condition here makes sense (applies to all modified godoc lines)

Comment (matthiasgubler) [reply staging/src/k8s.io/api/apps/v1/types.go:240]: > Does this accurately describe behavior for pods without readiness probes? What about pods that set spec.readinessGates?

Good point. I rechecked the code and in the file podutils.go the function IsPodReady basically  just checks the Ready Condition. In the pkg/kubelet/status/generate.go file is the function GeneratePodReadyCondition defined, which checks, that every container has the status ready and all readinessGates are true.

According to the KEP documentation https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/580-pod-readiness-gates is ready defined as "ready condition and all readiness gates are true". 

My suggestion would be to have a text like this:
```suggestion
	// readyReplicas is the number of pods created by this StatefulSet controller with a Ready Condition and that are passing all readinessGates.
```

Or would you suggest a rather generic description? Do you have another suggestion?

Comment (liggitt) [reply staging/src/k8s.io/api/apps/v1/types.go:240]: I wouldn't try to describe everything that goes into determining the Ready condition here... just limit it to saying it's the count of pods created by this controller with a Ready condition

Comment (matthiasgubler) [reply staging/src/k8s.io/api/apps/v1/types.go:240]: 👍
Next week I will commit a new version.

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 2]
```diff
@@ -237,7 +237,7 @@ type StatefulSetStatus struct {
 	// replicas is the number of Pods created by the StatefulSet controller.
 	Replicas int32 `json:"replicas" protobuf:"varint,2,opt,name=replicas"`
 
-	// readyReplicas is the number of Pods created by the StatefulSet controller that have a Ready Condition.
+	// readyReplicas is the number of pods created by this StatefulSet controller with a Ready Condition.
```
Comment (liggitt) [inline staging/src/k8s.io/api/apps/v1/types.go:240]:
```suggestion
	// readyReplicas is the number of pods created for this StatefulSet with a Ready Condition.
```

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 3]
```diff
@@ -451,7 +451,7 @@ type DeploymentStatus struct {
 	// +optional
 	UpdatedReplicas int32 `json:"updatedReplicas,omitempty" protobuf:"varint,3,opt,name=updatedReplicas"`
 
-	// Total number of ready pods targeted by this deployment.
+	// readyReplicas is the number of pods targeted by this Deployment controller with a Ready Condition.
```
Comment (liggitt) [inline staging/src/k8s.io/api/apps/v1/types.go:454]:
```suggestion
	// readyReplicas is the number of pods targeted by this Deployment with a Ready Condition.
```

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 4]
```diff
@@ -828,7 +828,7 @@ type ReplicaSetStatus struct {
 	// +optional
 	FullyLabeledReplicas int32 `json:"fullyLabeledReplicas,omitempty" protobuf:"varint,2,opt,name=fullyLabeledReplicas"`
 
-	// The number of ready replicas for this replica set.
+	// readyReplicas is the number of pods targeted by this ReplicaSet controller with a Ready Condition.
```
Comment (liggitt) [inline staging/src/k8s.io/api/apps/v1/types.go:831]:
```suggestion
	// readyReplicas is the number of pods targeted by this ReplicaSet with a Ready Condition.
```

**Review by liggitt — COMMENTED** [review]
drop "controller" from the doc... the API field is on an object, not a controller (applies to all versions)

**Pull request merged**